doc/doxygen/public/developer-faq.doxygen

```diff
@@ -0,0 +1,291 @@
+ This happens whenever the Build-System calls <code>CMake</code> (which can be quite often, e.g., after changes to <code>CMakeLists.txt</code>).
+
+ @subsection how_to_speed_up_the_compile_process How to speed up the compile process
+
+ To speed up the compile process of OpenMS, use several threads. If you have several processors/cores, build OpenMS
+ <code>classes/tests</code> and <code>TOPP</code> tools in several threads. On Linux, use the <code>make option -j: make -j8 OpenMS TOPP test_build</code>.
+
+ On Windows, Visual Studio solution files are automatically build with the <code>/MP</code> flag, such that Visual Studio uses all available cores of the machine.
+
+ @section continuous_integration Continuous integration
+
+ <h2><a class="anchor" id="how_does_travis_work"></a>How does <code>travis</code> work?</h2>
+
+ <a href="https://travis-ci.org/"><code>Travis</code></a> is an automated system for continuous integration and each new commit and pull request is automatically run through the <code>travis</code> build system. This is controlled by a <code>.travis.yaml</code> file in the source tree.
+
+ <h2><a class="anchor" id="what_can_we_do_if_travis_times_out"></a>What can we do if <code>travis</code> times out?</h2>
+
+ First, restart <code>travis</code>, as it sometimes hangs. Since <code>travis</code> builds on shared infrastructure, the next build may work better. This needs to be done by an OpenMS core developer.
+
+ Since we use extensive caching, the build may take much longer when many files are touched and may never complete (running into the <code>travis</code> time limit). In that case, we can rebuild the cache using the following approach:
+
+ @code{sh}
+ $ git cherry-pick 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+ $ git push origin develop
+ # Now wait for the build to complete
+ $ git revert 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+ $ git push origin develop
+ @endcode
+
+ @section testing Testing
+
+ @subsection how_to_update_many_test_files How to update many test files
+
+ Use <code>grep</code> to extract the lines starting with <code>diff FILENAME1 FILENAME2</code> and replace the <code>diff</code> by copy.
+
+ @section working_in_ides Working in IDEs
+
+ @subsection why_there_are_no_source_test_and_source_app_topp_utils_folders Why are there no source/TEST and source/APPLICATIONS/TOPP|UTILS folder?
+
+ All source files added to an IDE are associated with their targets. Find the source files for each test within
+ its own subproject. The same is true for the <code>TOPP</code> and <code>UTILS</code> classes.
+
+ @subsection im_getting_the_error_error_c2471 I'm getting the error "Error C2471: cannot update program database"
+
+ This is a bug in Visual Studio and there is a <a href="http://code.msdn.microsoft.com/KB946040">bug fix</a>. Only apply it if you encounter the error. The bug fix might have unwanted side effects!
```

Comment:
this link is a 404, let's ask/confirm in discord.

Reply:
I think it is meant to be: https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2471?view=msvc-170